Networks.yaml

```diff
@@ -0,0 +1,146 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "Network Infrastructure - VPC, Subnets, NAT Gateways, and Route Tables"
+
+Parameters:
+  VpcCidrBlock:
+    Description: CIDR block for the VPC
+    Type: String
+    Default: 172.16.0.0/16
+    AllowedPattern: ^(([0-9]{1,3}\.){3}[0-9]{1,3}\/([0-9]|[1-2][0-9]|3[0-2]))$
+    ConstraintDescription: Must be a valid CIDR block (e.g., 172.16.0.0/16).
+
+Resources:
+  # VPC
+  MainVPC:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: !Ref VpcCidrBlock
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: Main-VPC
+
+  # Internet Gateway
+  InternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: Main-IGW
+
+  AttachGateway:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId: !Ref MainVPC
+      InternetGatewayId: !Ref InternetGateway
+
+  # Subnets
+  PublicSubnet1:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref MainVPC
+      CidrBlock: 172.16.0.0/24
+      AvailabilityZone: eu-north-1a
```

Comment:
Using static value like this will prevent you from applying the same IAC on other regions. Use select func to get AZ. example, `!Select [ 0, !GetAZs '' ]`